event-hubs.api.md

```diff
@@ -0,0 +1,303 @@
+import { SharedKeyCredential } from '@azure/core-amqp';
+import { Span } from '@azure/core-tracing';
+import { SpanContext } from '@azure/core-tracing';
+import { SpanOptions } from '@azure/core-tracing';
+import { TokenCredential } from '@azure/core-amqp';
+import { TokenType } from '@azure/core-amqp';
+import { WebSocketImpl } from 'rhea-promise';
+
+// @public
+export interface AbortSignalOptions {
+    abortSignal?: AbortSignalLike;
+}
+
+// @public
+export interface Checkpoint {
+    consumerGroupName: string;
+    eTag: string;
+    eventHubName: string;
+    fullyQualifiedNamespace: string;
+    offset: number;
+    ownerId: string;
+    partitionId: string;
+    sequenceNumber: number;
+}
+
+// @public
+export interface CheckpointManager {
+    listCheckpoints(fullyQualifiedNamespace: string, eventHubName: string, consumerGroup: string): Promise<Checkpoint[]>;
+    updateCheckpoint(checkpoint: Checkpoint): Promise<string>;
+}
+
+// @public
+export enum CloseReason {
+    OwnershipLost = "OwnershipLost",
+    Shutdown = "Shutdown"
+}
+
+// @public
+export interface CreateBatchOptions {
+    abortSignal?: AbortSignalLike;
+    maxSizeInBytes?: number;
+    partitionKey?: string;
+}
+
+export { DataTransformer }
+
+export { DefaultDataTransformer }
+
+// @public
+export interface EventData {
+    body: any;
+    properties?: {
+        [key: string]: any;
+    };
+}
+
+// @public
+export class EventDataBatch {
+    // Warning: (ae-forgotten-export) The symbol "ConnectionContext" needs to be exported by the entry point index.d.ts
+    // 
+    // @internal
+    constructor(context: ConnectionContext, maxSizeInBytes: number, partitionKey?: string);
+    readonly batchMessage: Buffer | undefined;
+    readonly count: number;
+    // @internal
+    readonly _messageSpanContexts: SpanContext[];
+    readonly partitionKey: string | undefined;
+    readonly sizeInBytes: number;
+    tryAdd(eventData: EventData, options?: TryAddOptions): boolean;
+}
+
+// @public
+export interface EventHubClientOptions {
+    dataTransformer?: DataTransformer;
+    retryOptions?: RetryOptions;
+    userAgent?: string;
+    webSocket?: WebSocketImpl;
+    webSocketConstructorOptions?: any;
+}
+
+// @public
+export class EventHubConsumerClient {
+    constructor(connectionString: string, options?: EventHubClientOptions);
+    constructor(connectionString: string, eventHubName: string, options?: EventHubClientOptions);
+    constructor(host: string, eventHubName: string, credential: TokenCredential, options?: EventHubClientOptions);
+    close(): Promise<void>;
+    static defaultConsumerGroupName: string;
+    getPartitionIds(option?: GetPartitionIdsOptions): Promise<string[]>;
+    getPartitionProperties(partitionId: string, options?: GetPartitionPropertiesOptions): Promise<PartitionProperties>;
+    getProperties(options?: GetPropertiesOptions): Promise<EventHubProperties>;
+    subscribe(consumerGroupName: string, options: SubscriptionOptions): Subscription;
```

Comment:
Consider `SubscriptionEventHandlers` to be a separate argument so that we don't have an "options" that is not optional

```
subscribe(consumerGroupName: string, handlers: SubscriptionEventHandlers, options: SubscriptionOptions)
```

Reply:
+1 (and note if we do this, options will become `options?: SubscriptionOptions`, since no longer required)

Reply:
Do you allow users to set ownerLevel or hardcode 0?